Sources/Apollo/ApolloClient.swift

```diff
@@ -281,9 +284,13 @@ private final class FetchQueryOperation<Query: GraphQLQuery>: AsynchronousOperat
   }
 
   func fetchFromNetwork() {
+    networkTask?.cancel()
     networkTask = client?.send(operation: query,
                                shouldPublishResultToStore: true,
-                               context: context) { result in
+                               context: context) { [weak self] result in
+      guard let self = self else {
+        return
+      }
```

Comment:
I don't think this and the above `weak self` guards are necessary because the operation will be held on to by the operation queue, so it can't get deallocated before it finishes or cancels (and below we're cancelling the network task if the operation is cancelled).

Reply:
but don't you create circular retain here? FetchQueryOperation -> client -> FetchQueryOperation

Reply:
🤔Yeah I suppose that would create a retain cycle if there's a cancellation. Fair point!